packages/ember-views/lib/components/component.js

```diff
@@ -158,6 +158,22 @@ var Component = View.extend(TargetActionSupport, {
 
       this.layout = this.defaultLayout;
     }
+
+    // If in a tagless component, assert that no event handlers are defined
+    assert(
+      `You can not define a function that handles DOM events in the \`${this}\` tagless component since it doesn't have any DOM element.`,
+      this.tagName !== '' || !(() => {
+        var eventDispatcher = this.container.lookup('event_dispatcher:main');
+
+        for (var key in eventDispatcher._finalEvents) {
```

Comment:
> we don't throw if eventDispatcher._finalEvents is undefined or if eventDispatcher itself is undefined

Correct.

> Do we need to test that case?

Wouldn't hurt to have a test, but I'm not super worried about testing it.